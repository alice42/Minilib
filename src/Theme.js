export const theme = {
  backgroundColor: '#ECEFF1',
  onBackground: '#000000',
  surfaceColor: '#FFFFFF',
  surfaceColorHover: '#F5F5F5',
  surfaceColorActive: '#E0E0E0',
  onSurface: '#000000',
  primaryColor: '#0277BD',
  primaryColorHover: '#0288D1',
  primaryColorActive: '#01579B',
  onPrimary: '#FFFFFF',
  secondaryColor: '#00897B',
  secondaryColorHover: '#00695C',
  secondaryColorActive: '#004D40',
  onSecondary: '#FFFFFF',
  tertiaryColor: '#EF6C00',
  tertiaryColorHover: '#FB8C00',
  tertiaryColorActive: '#E65100',
  onTertiary: '#FFFFFF',
  fontPrimary: {
    regular: 'Lato,sans-serif',
    bold: 'Lato,sans-serif',
    light: 'Lato,sans-serif'
  },
  fontSecondary: {
    regular: 'arial,sans-serif',
    bold: 'arial,sans-serif',
    light: 'arial,sans-serif'
  },
  fontTertiary: {
    regular: 'Georgia,serif',
    bold: 'Georgia,serif',
    light: 'Georgia,serif'
  },
  font48: '4rem',
  font43: '3.6rem',
  font38: '3.2rem',
  font34: '2.9rem',
  font30: '2.5rem',
  font29: '2.4rem',
  font28: '2.3rem',
  font26: '2.1rem',
  font24: '2rem',
  font23: '1.9rem',
  font22: '1.85rem',
  font20: '1.65rem',
  font19: '1.6rem',
  font18: '1.5rem',
  font17: '1.4rem',
  font16: '1.35rem',
  font15: '1.25rem',
  font14: '1.2rem',
  font13: '1.1rem',
  font12: '1rem',
  font11: '0.95rem',
  font10: '0.9rem',
  font9: '0.75rem',
  font8: '0.7rem',
  font6: '0.5rem',
  overlayOpacity: 0.7,
  borderRadius: '5px',
  border: 'solid 5px #000000'
}
