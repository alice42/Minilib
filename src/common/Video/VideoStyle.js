import styled, { css } from 'styled-components'

const VideoStyle = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  ${({ width }) => width && css`
    width: ${width};
  `}
  & > div:first-of-type {
    display: flex;
    overflow: hidden;
    ${({ border, theme }) => border && css`
      box-sizing: border-box;
      border: ${typeof border === 'string' ? border : theme.border};
    `}
    ${({ rounded, theme }) => rounded && css`
      border-radius: ${typeof rounded === 'string' ? rounded : theme.borderRadius};
    `}
    ${({ height }) => height && css`
      height: ${height};
    `}

    & > video {
      width: ${({ cover, filled, contain }) => cover || filled || contain ? '100%' : 'unset'};
      filter: ${({ blur }) => 'blur(' + blur + ')'};
      object-fit: ${({ cover, filled, contain }) => filled ? 'fill' : contain ? 'contain' : cover ? 'cover' : 'unset'};
    }
  }
  ${({ customStyle }) => css(customStyle)}
`

export default VideoStyle
