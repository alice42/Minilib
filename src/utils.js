export let conflictsValidator = (validType, conflicts) => (props, propName, componentName) => {
  let conflictedProp
  if (typeof props[propName] !== validType) return new Error(`<${componentName}>: ${propName} must be a ${validType}!`)
  if (props[propName] && (conflictedProp = conflicts.find(p => props[p]))) {
    return new Error(`<${componentName}>: ${propName} can't be used if ${conflictedProp} is already used!`)
  }
}
