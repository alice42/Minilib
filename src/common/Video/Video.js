import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Overlay from '../Overlay/Overlay'
import VideoStyle from './VideoStyle'
import { conflictsValidator } from '../../utils'

class Video extends Component {
  static propTypes = {
    /** link to the image file ("./picture.jpg") */
    src: PropTypes.string.isRequired,
    /** link to the poster (image while loading) */
    poster: PropTypes.string,
    /** enabled controls */
    controls: PropTypes.bool,
    /** enabled autoPlay */
    autoplay: PropTypes.bool,
    /** mute video audio */
    muted: PropTypes.bool,
    /** enabled loop */
    loop: PropTypes.bool,
    /** round borders style on image (ex: true, "5px") */
    rounded: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** video borders ("solid 5px #000000") */
    border: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** object-fit: cover (if :=false => 'none') */
    cover: conflictsValidator('boolean', ['contain', 'filled']),
    /** object-fit: contain */
    contain: conflictsValidator('boolean', ['cover', 'filled']),
    /** object-fit: fill */
    filled: conflictsValidator('boolean', ['contain', 'cover']),
    /** add a blur effect on the image ("10px", "1.2em", ...) */
    blur: PropTypes.string,
    /** height of the image ("200px", "30%", "45vh", ...) */
    height: PropTypes.string,
    /** width of the image ("200px", "30%", "45vw", ...) */
    width: PropTypes.string,
    /** pass props to the overlay, <Image>{ overlay content }</Image> */
    overlay: PropTypes.object,
    /** custom style object */
    customStyle: PropTypes.object
  }
  static defaultProps = {
    rounded: false,
    controls: true,
    autoplay: false,
    muted: false,
    loop: false,
    cover: false,
    contain: false,
    filled: false,
    border: false,
    blur: '0px',
    customStyle: {}
  }

  render () {
    const { src, title, overlay, controls, autoplay, muted, loop, poster } = this.props
    const alt = src.indexOf('/') !== -1 ? src.slice(src.lastIndexOf('/') + 1) : src
    return (
      <VideoStyle {...this.props}>
        <div>
          <video
            alt={alt}
            title={title || (alt.indexOf('.') !== -1 ? alt.slice(0, alt.lastIndexOf('.')) : alt)}
            controls={controls}
            autoPlay={autoplay}
            muted={muted}
            loop={loop}
            poster={poster}
          >
            HTML5 video unsupported by your browser
            <source src={src} type={'video/' + src.slice(src.lastIndexOf('.') + 1).toLowerCase()} />
          </video>
        </div>
        {
          this.props.children || (overlay && Object.keys(overlay).length) ? <Overlay {...overlay}>{ this.props.children }</Overlay> : null
        }
      </VideoStyle>
    )
  }
}

export default Video
