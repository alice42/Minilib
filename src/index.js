import Button from './common/Buttons/Button'
import Icon from './common/Icons/Icon'
import Image from './common/Image/Image'
import InputText from './common/Inputs/InputText/InputText'
import ExternalLink from './common/Links/ExternalLink'
import Overlay from './common/Overlay/Overlay'
import Video from './common/Video/Video'

export { Button }
export { Icon }
export { Image }
export { InputText }
export { ExternalLink }
export { Overlay }
export { Video }
