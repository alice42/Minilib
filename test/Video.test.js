import React from 'react'
import Video from '../src/common/Video/Video'
import 'jest-styled-components'
import { renderCreateWithThemeJSON } from './TestUtils'
import { theme } from '../src/Theme'

const src = '/video/video.mp4'

test(':src', () => {
  const comps = [
    renderCreateWithThemeJSON(<Video src={src} />)
  ]
  expect(Video).toThrow()
  expect(comps[0].children).not.toBeNull()
  expect(comps[0].children[0]).not.toBeNull()
  expect(comps[0].children[0].children[0].type).toEqual('video')
  expect(comps[0].children[0].children[0].children[1].type).toEqual('source') // children[1] because of the no HTML5 browser display message in children[0]
  expect(comps[0].children[0].children[0].children[1].props.src).toEqual(src)
  expect(comps[0]).toMatchSnapshot()
})

test(':height & :width', () => {
  const comps = [
    renderCreateWithThemeJSON(<Video src={src} height={'200px'} width={'500px'} />),
    renderCreateWithThemeJSON(<Video src={src} height={'200px'} />),
    renderCreateWithThemeJSON(<Video src={src} width={'500px'} />)
  ]
  expect(comps[0]).toHaveStyleRule('height', '200px', { modifier: '& > div:first-of-type' })
  expect(comps[0]).toHaveStyleRule('width', '500px')
  expect(comps[0]).toHaveStyleRule('width', 'unset', { modifier: '& > div:first-of-type > video' })
  expect(comps[1]).toHaveStyleRule('height', '200px', { modifier: '& > div:first-of-type' })
  expect(comps[1]).not.toHaveStyleRule('width')
  expect(comps[2]).toHaveStyleRule('width', '500px')
  expect(comps[2]).not.toHaveStyleRule('height', { modifier: '& > div:first-of-type' })
  expect(comps[0]).toMatchSnapshot()
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toMatchSnapshot()
})

test(':object-fit', () => {
  const comps = [
    renderCreateWithThemeJSON(<Video src={src} />),
    renderCreateWithThemeJSON(<Video src={src} cover />),
    renderCreateWithThemeJSON(<Video src={src} contain />),
    renderCreateWithThemeJSON(<Video src={src} filled />),
    renderCreateWithThemeJSON(<Video src={src} cover={false} />)
  ]
  expect(comps[0]).toHaveStyleRule('object-fit', 'unset', { modifier: '& > div:first-of-type > video' })
  expect(comps[1]).toHaveStyleRule('object-fit', 'cover', { modifier: '& > div:first-of-type > video' })
  expect(comps[2]).toHaveStyleRule('object-fit', 'contain', { modifier: '& > div:first-of-type > video' })
  expect(comps[3]).toHaveStyleRule('object-fit', 'fill', { modifier: '& > div:first-of-type > video' })
  expect(comps[4]).toHaveStyleRule('object-fit', 'unset', { modifier: '& > div:first-of-type > video' })
  expect(comps[0]).toMatchSnapshot()
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toMatchSnapshot()
  expect(comps[3]).toMatchSnapshot()
  expect(comps[4]).toMatchSnapshot()
})

test(':rounded', () => {
  const comps = [
    renderCreateWithThemeJSON(<Video src={src} />),
    renderCreateWithThemeJSON(<Video src={src} rounded />),
    renderCreateWithThemeJSON(<Video src={src} rounded={'87px'} />)
  ]
  expect(comps[0]).not.toHaveStyleRule('border-radius', { modifier: '& > div:first-of-type' })
  expect(comps[1]).toHaveStyleRule('border-radius', theme.borderRadius, { modifier: '& > div:first-of-type' })
  expect(comps[2]).toHaveStyleRule('border-radius', '87px', { modifier: '& > div:first-of-type' })
  expect(comps[1]).toMatchSnapshot()
})

test(':filter', () => {
  const comps = [
    renderCreateWithThemeJSON(<Video src={src} asBackground blur={'5px'} />)
  ]
  expect(comps[0]).toHaveStyleRule('filter', `blur(5px)`, { modifier: '& > div:first-of-type > video' })
  expect(comps[0]).toMatchSnapshot()
})

test(':title & :alt', () => {
  const comps = [
    renderCreateWithThemeJSON(<Video src={src} />),
    renderCreateWithThemeJSON(<Video src={src} title={'HelloWorld!'} />),
    renderCreateWithThemeJSON(<Video src={'noSlashAndNoPoint'} />)
  ]
  expect(comps[0].children[0].children[0].props.title).toEqual('video')
  expect(comps[0].children[0].children[0].props.alt).toEqual('video.mp4')
  expect(comps[1].children[0].children[0].props.title).toEqual('HelloWorld!')
  expect(comps[1].children[0].children[0].props.alt).toEqual('video.mp4')
  expect(comps[2].children[0].children[0].props.alt).toEqual('noSlashAndNoPoint')
  expect(comps[2].children[0].children[0].props.title).toEqual('noSlashAndNoPoint')
})

test(':border', () => {
  const comps = [
    renderCreateWithThemeJSON(<Video src={src}>Hello</Video>),
    renderCreateWithThemeJSON(<Video src={src} border>Hello</Video>),
    renderCreateWithThemeJSON(<Video src={src} border={'dashed 2px red'} />)
  ]
  expect(comps[0]).not.toHaveStyleRule('border', { modifier: '& > div:first-of-type' })
  expect(comps[0]).not.toHaveStyleRule('box-sizing', { modifier: '& > div:first-of-type' })
  expect(comps[1]).toHaveStyleRule('border', theme.border, { modifier: '& > div:first-of-type' })
  expect(comps[1]).toHaveStyleRule('box-sizing', 'border-box', { modifier: '& > div:first-of-type' })
  expect(comps[2]).toHaveStyleRule('border', 'dashed 2px red', { modifier: '& > div:first-of-type' })
  expect(comps[2]).toHaveStyleRule('box-sizing', 'border-box', { modifier: '& > div:first-of-type' })
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toMatchSnapshot()
})

test(':overlay', () => {
  const comps = [
    renderCreateWithThemeJSON(<Video src={src}>Hello</Video>),
    renderCreateWithThemeJSON(<Video src={src} overlay={{ dark: true }} />)
  ]
  expect(comps[0].children.length).toEqual(2)
  expect(comps[1].children.length).toEqual(2)
  expect(comps[0]).toMatchSnapshot()
  expect(comps[1]).toMatchSnapshot()
})
