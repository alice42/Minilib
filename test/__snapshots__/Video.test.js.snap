// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:border 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  box-sizing: border-box;
  border: solid 5px #000000;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`:border 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  box-sizing: border-box;
  border: dashed 2px red;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:filter 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(5px);
  filter: blur(5px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:height & :width 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 500px;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  height: 200px;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  height="200px"
  loop={false}
  muted={false}
  src="/video/video.mp4"
  width="500px"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:height & :width 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  height: 200px;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  height="200px"
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:height & :width 3`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 500px;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
  width="500px"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:object-fit 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:object-fit 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: 100%;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: cover;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:object-fit 3`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: 100%;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: contain;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:object-fit 4`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: 100%;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: fill;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:object-fit 5`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:overlay 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`:overlay 2`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #ECEFF1;
  background-color: rgba(0,0,0,0.7);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
  <div
    className="c1"
  />
</div>
`;

exports[`:rounded 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  border-radius: 5px;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`:src 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div:first-of-type {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c0 > div:first-of-type > video {
  width: unset;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  controls={true}
  loop={false}
  muted={false}
  src="/video/video.mp4"
>
  <div>
    <video
      alt="video.mp4"
      autoPlay={false}
      controls={true}
      loop={false}
      muted={false}
      title="video"
    >
      HTML5 video unsupported by your browser
      <source
        src="/video/video.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;
